R/helper_lm_robust_fit.R

```diff
@@ -535,8 +530,24 @@ get_fstat <- function(tss_r2s,
   return(f)
 }
 
-compute_fstat <- function(coefs, vcov_fit, nomdf) {
-  crossprod(coefs, chol2inv(chol(vcov_fit)) %*% coefs) / nomdf
+compute_fstat <- function(coef_matrix, coef_indices, vcov_fit, rank, nomdf) {
+  fstat <- tryCatch({
+    vapply(seq_len(ncol(coef_matrix)),
+           function(x) {
+             vcov_indices <- coef_indices + (x - 1) * rank
+             crossprod(
+               coef_matrix[coef_indices, x],
+               chol2inv(chol(vcov_fit[vcov_indices, vcov_indices])) %*%
+                 coef_matrix[coef_indices, x]
+             ) / nomdf
+           },
+           numeric(1))
+  },
+  error = function(e) {
+    rep(NA_real_, length(ncol(coef_matrix)))
```

Comment:
If one fails, the whole thing is NA?

Reply:
Hmmm, good point. Originally was written with only one outcome. Will fix and test.